manager/controllers/app/moduleinstance.go

```diff
@@ -277,6 +289,9 @@ func (m *ModuleManager) SelectModuleInstances(item modules.DataInfo, appContext
 			m.Log.Info("Allocation failed: " + err.Error())
 			return instances, err
 		}
+                copyCluster, vaultAuthMethod, err := copySelector.SelectCluster(item, m.Clusters)
```

Comment:
1. Why moved here? The err value is not checked.
2. vaultAuthMethod should not be returned by SelectCluster. Once you have a cluster, you can iterate over clusters to get its property.

Reply:
It is moved before the module arguments are set. I'll add a check and iteration.